### DISCUSSION TITLE
T0 - Parte 2 (TRADE-PUFFLE y GIVE-PUFFLES)

### DISCUSSION THREAD

**Discussion by SantiagoOsorio3** [IIC2133-PUC/2024-2#19] [Tarea 0] [answered]
Hola, tengo una duda con respecto a como fuciona TRADE-PUFFLE y GIVE-PUFFLES en conjunto, ya que segun enunciado hay que respectar lo siguiente en TRADE-PUFFLE:

Se deben respetar las posiciones de los respectivos inventarios. Esto implica que, si el pingüino ID_penguin_1 intercambia su puffle ID_puffle_1 (que había llegado en segundo lugar a su posición en el inventario) con el puffle ID_puffle_2 del pingüino ID_penguin_2 (que estaba en la novena posición en el inventario de su dueño), se debe cumplir que después del intercambio el puffle ID_puffle_2 pase a ocupar la segunda posición en la lista del pingüino ID_penguin_1, mientras que el puffle con ID_puffle_1 ocupe la novena posición en la lista del pingüino ID_penguin_2.

SIn embargo, que pasa si justo despues de hacer el TRADE-PUFFLE desde la posicion 0 (osea que esta posicion despues del trade queda libre) se hace un GIVE-PUFFLES. En este caso deberia antes de hacer el GIVE-PUFFLES un ordenamiento de los puffles (como por ejemplo si tenia puffles en las posiciones 1,3,6, hacer que se muevan a las posiciones  0,1,2) para que cuando agregue o se agregen los puffles nuevos de GIVE-PUFFLES no queden en la posicion 0,2,4 por ejemplo y queden en la posicion 3,....

Tengo esta duda ya que me esta pasando algo de este estilo en un ejemplo.

**Comment by sebacarrascop** [comment, marked as answer]
> osea que esta posicion despues del trade queda libre 

El trade puffle como mencionas en tu segundo parrado nunca deja "posiciones libres", ya que es un intercambio de puffle, donde además intercambian de posiciones. Por lo que en el caso que mencionas, al hacer el GIVE, se entregaría el puffle que acaba de recibir en el intercambio del evento anterior.
No sé si queda claro, quizás no entendí bien tu pregunta, quedo atento.
